resources/adoptopenjdk_install.rb

```diff
@@ -0,0 +1,178 @@
+      package 'tar' do
+        not_if { platform_family?('mac_os_x', 'windows') }
+        action :nothing
+      end.run_action(:install)
+
+      cmd = shell_out(%(tar xvzf "#{Chef::Config[:file_cache_path]}/#{tarball_name}" -C "#{Chef::Config[:file_cache_path]}" --no-same-owner)
+                     )
+      unless cmd.exitstatus == 0
+        Chef::Application.fatal!("Failed to extract file #{tarball_name}!")
+      end
+
+      cmd = shell_out(
+        %(mv "#{Chef::Config[:file_cache_path]}/#{app_dir_name}" "#{app_dir}" )
+      )
+      unless cmd.exitstatus == 0
+        Chef::Application.fatal!(%( Command \' mv "#{Chef::Config[:file_cache_path]}/#{app_dir_name}" "#{app_dir}" \' failed ))
+      end
+
+      # change ownership of extracted files
+      FileUtils.chown_R new_resource.owner, app_group, app_dir
+    end
+  end
+
+  # set up .jinfo file for update-java-alternatives
+  java_name = app_home.split('/')[-1]
+  jinfo_file = "#{app_root}/.#{java_name}.jinfo"
+  if platform_family?('debian') && !::File.exist?(jinfo_file)
+    converge_by("Add #{jinfo_file} for debian") do
+      template jinfo_file do
+        cookbook 'java'
+        source 'adoptopenjdk.jinfo.erb'
+        owner new_resource.owner
+        group app_group
+        variables(
+          priority: new_resource.alternatives_priority,
+          bin_cmds: new_resource.bin_cmds,
+          name: java_name,
+          app_dir: app_home
+        )
+        action :nothing
+      end.run_action(:create)
+    end
+  end
+
+  # link app_home to app_dir
+  Chef::Log.debug "app_home is #{app_home} and app_dir is #{app_dir}"
+  current_link = ::File.symlink?(app_home) ? ::File.readlink(app_home) : nil
+  if current_link != app_dir
+    converge_by("symlink #{app_dir} to #{app_home}") do
+      FileUtils.rm_f app_home
+      FileUtils.ln_sf app_dir, app_home
+      FileUtils.chown new_resource.owner, app_group, app_home
+    end
+  end
+
+  # update-alternatives
+  java_alternatives 'set-java-alternatives' do
+    java_location app_home
+    bin_cmds new_resource.bin_cmds
+    priority new_resource.alternatives_priority
+    default new_resource.default
+    reset_alternatives new_resource.reset_alternatives
+    action :set
+  end
+end
+
+action :remove do
+  app_dir_name, _tarball_name = parse_app_dir_name(new_resource.url)
+  app_root = new_resource.app_home.split('/')[0..-2].join('/')
+  app_dir = app_root + '/' + app_dir_name + '-' + new_resource.variant
+  app_home = new_resource.app_home
+  Chef::Log.debug("processing #{new_resource.variant} variant")
+
+  if ::File.exist?(app_dir)
+    java_alternatives 'unset-java-alternatives' do
+      java_location app_home
+      bin_cmds new_resource.bin_cmds
+      action :unset
+    end
+
+    converge_by("remove #{new_resource.name} at #{app_dir}") do
+      Chef::Log.info "Removing #{new_resource.name} at #{app_dir}"
+      FileUtils.rm_rf app_dir
```

Comment:
is there some reason to not use the `directory` resource with an action of `:delete` and `recursive` set to `true`? Is it for the force flag for some edgecase?